extensions/geode-modules/src/main/java/org/apache/geode/modules/session/catalina/ClientServerSessionCache.java

```diff
@@ -247,12 +246,9 @@ void createSessionRegionOnServers() {
     // Create the region
     Region region = factory.create(getSessionManager().getRegionName());
 
-    /*
-     * If we're using an empty client region, we register interest so that expired sessions are
-     * destroyed correctly.
-     */
-    if (!getSessionManager().getEnableLocalCache()) {
-      region.registerInterest("ALL_KEYS", InterestResultPolicy.KEYS);
+    // register interest to get updates from server if local cache is enabled
+    if (getSessionManager().getEnableLocalCache()) {
```

Comment:
Again, we flipped the logic.  Will this cause unexpected behavior changes?

Reply:
Yes, we need both.